crawl-ref/source/dat/des/branches/lair.des

```diff
@@ -453,12 +453,11 @@ KMONS:   N = emperor scorpion / orb spider
 KFEAT:   ` = floor
 NSUBST:  A = 1:A / *:W
 KMONS:   A = kraken / alligator snapping turtle
-KFEAT:   Aa = shallow_water
+KFEAT:   A = shallow_water
 NSUBST:  S = 1:S / *:W
 KMONS:   S = bunyip / goliath frog
-KFEAT:   Ss = shallow_water
+KFEAT:   S = shallow_water
 NSUBST:  L = 1:L / *:`
-SUBST:   L = Ll
 KMONS:   L = catoblepas / eight-headed hydra
 KFEAT:   ` = floor
```

Comment:
There's a lot of redundancy here. 
One thing to keep in mind is that KFEAT/KMONS/KMASK/KITEM/ITEM/MONS etc. apply only after all the various SUBST and SHUFFLE and NSUBST have happened, even ones that are defined after them in the vault definition. (KPROP and the various tile/colour changing commands are the odd ones out here; they only take into account the substitutions that are defined before them.) You don't need to have three separate lines saying KFEAT: ` = floor. (In fact, because ` doesn't have any meaning on its own, the feature there will be set to floor by default. So you don't even really need those KFEAT lines at all.) Because of this, it's also easier to simply have all your K-statements in one part, then all the substitution statements in another part, rather than mixing them around like you've done here.

You can also combine NSUBST lines into one:
NSUBST: P = 1:P / *:`, N = 1:N / *:`, A = 1:A / *:W, S = 1:S / *:W, L = 1:L / *:` 
which can be shortened even further:
NSUBST: P = P / `, N = N / `, A = A / W, S = S / W, L = L / `
since if no numbers are specified for the replacement glyphs, the last one will be interpreted to have *= in front of it and the rest will be interpreted to have 1=.

Likewise, you can combine the KFEAT lines:
KFEAT: AS = shallow_water